File: web/src/__generated/services/CloudController.ts
import type {Executor} from '../';
import type {CloudSpaceInfo} from '../model/static/';

/**
 * 云端控制器
 * 提供云端相关功能 API
 */
export class CloudController {
    
    constructor(private executor: Executor) {}
    
    /**
     * 获取云端空间使用情况
     * 
     * @parameter {CloudControllerOptions['getCloudSpace']} options
     * - accountId 账号 ID
     * @return 云端空间信息
     * 
     */
    readonly getCloudSpace: (options: CloudControllerOptions['getCloudSpace']) => Promise<
        CloudSpaceInfo
    > = async(options) => {
        let _uri = '/api/cloud/space/';
        _uri += encodeURIComponent(options.accountId);
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<CloudSpaceInfo>;
    }
}

export type CloudControllerOptions = {
    'getCloudSpace': {
        /**
         * 账号 ID
         */
        readonly accountId: number
    }
}

File: server/src/main/kotlin/com/coooolfan/xiaomialbumsyncer/controller/CloudController.kt
package com.coooolfan.xiaomialbumsyncer.controller

import cn.dev33.satoken.annotation.SaCheckLogin
import com.coooolfan.xiaomialbumsyncer.xiaomicloud.CloudSpaceInfo
import com.coooolfan.xiaomialbumsyncer.xiaomicloud.XiaoMiApi
import org.babyfish.jimmer.client.meta.Api
import org.noear.solon.annotation.Controller
import org.noear.solon.annotation.Managed
import org.noear.solon.annotation.Mapping
import org.noear.solon.annotation.Path
import org.noear.solon.core.handle.MethodType

/**
 * 云端控制器
 * 提供云端相关功能 API
 */
@Api
@Managed
@Mapping("/api/cloud")
@Controller
@SaCheckLogin
class CloudController(private val xiaoMiApi: XiaoMiApi) {

    /**
     * 获取云端空间使用情况
     *
     * @param accountId 账号 ID
     * @return 云端空间信息
     *
     * @api GET /api/cloud/space/{accountId}
     * @permission 需要登录认证
     */
    @Api
    @Mapping("/space/{accountId}", method = [MethodType.GET])
    fun getCloudSpace(@Path accountId: Long): CloudSpaceInfo {
        return xiaoMiApi.getCloudSpace(accountId)
    }
}
